### PR TITLE
Tree shaking support

### PR DESCRIPTION
**Pull request by georgesmith46**
Fixes: #181 #180

# Changes
- Changed the build tool from Webpack to rollup. The main build output is now using the UMD format, this is very similar to CJS except it has better script loader support.
- Added ESM build output.
- Changed from Webpack to Parcel for building the demo app. Originally, I tried using rollup for this to keep things simple but it ended up being more complicated. I decided to go with Parcel over Webpack purely because of its zero-config approach.
- Moved the demo app files into one folder.
- The demo app HTML originally used CDN scripts for React v16. Parcel bundles all dependencies together, including React, so I've removed these CDN scripts.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -97,5 +98,10 @@
       "js",
       "jsx"
     ]
+  },
+  "targets": {
```
Comment (georgesmith46) [inline package.json:102]: This is to inform Parcel that it can ignore the package.json since we're building a demo app

File: package.json [hunk 2]
```diff
@@ -64,21 +67,19 @@
...
-    "html-webpack-plugin": "^5.3.1",
     "husky": "^6.0.0",
     "jest": "^26.6.3",
     "lint-staged": "^11.0.0",
+    "parcel": "^2.0.0-beta.2",
     "prettier": "^2.3.0",
     "react": "^17.0.2",
     "react-dom": "17.0.2",
-    "react-hot-loader": "^4.13.0",
+    "rollup": "^2.52.7",
```
Comment (cchanxzy) [inline package.json:77]: Is rollup required for parcel? Or is this leftover?

**Comment by cchanxzy** [comment]
Thanks @georgesmith46 !

Love the changes and everything feels much more simplified and tidied up 👍 

I haven't used parcel before, but it looks pretty cool.

I just had that one question, but pretty happy to merge soon, everything else looks good.

File: package.json (hunk 2 above)
Comment (georgesmith46) [reply package.json:77]: So the way I've done it is rollup will be used to build the package, and parcel will be used to start/build the demo app.

It doesn't necessarily need to be these tools, it's just that they provide the simplest approach.

Comment (cchanxzy) [reply package.json:77]: Right, just wanted to make sure it was required. Thanks

**Pull request merged**